pmlc/rt/boundscheck.cc

```diff
@@ -0,0 +1,16 @@
+// Copyright 2020 Intel Corporation
+
+#include <stdint.h>
+
+#include <stdexcept>
+
+// This function is called from the BoundsCheckPass.
+// index - ndex for a dimension that is accessed by load/store operation
+// range - upper bound of the range for this dimension (it is always 0 -
+// range)
+extern "C" void plaidml_rt_bounds_check(intptr_t index, unsigned range) {
```

Comment:
Why are we using `unsigned` for the `range`? Doesn't that mean the bitwidth is machine-specific? I would have expected an `int64_t` here instead.

Reply:
It was coming like this from the codegen side. Fixed to use int64_t now.

Reply:
So I think @flaub's point was that we shouldn't make any of our extern C functions take any parameters that are bitwidth specific (i.e. different for 64 bit or 32 bit compilers), which of course includes intptr_t as well.  Let's always used completely fixed bitwidth types for all interfacing.

Reply:
What I meant was that we shouldn't use `unsigned` in the C signature.